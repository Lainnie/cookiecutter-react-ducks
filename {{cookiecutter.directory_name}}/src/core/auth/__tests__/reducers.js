import authReducers, { authInitialStates, authOperations } from '../../auth';

describe('auth reducers', () => {
  describe('auth/LOGIN_SUCCESS', () => {
    const setItem = localStorage.setItem;
    beforeEach(() => {
      localStorage.setItem = jest.fn();
    });

    afterEach(() => {
      localStorage.setItem = setItem;
    });

    it.only('updates username and accessToken', () => {
      const state = {
        ...authInitialStates,
        login: {
          ...authInitialStates.login,
          error: true, // Simulate a failed attempt
        },
      };
      const newState = authReducers(
        state,
        authOperations.loginSuccess({
          username: 'admin',
          accessToken: '1234567890',
        })
      );

      expect(newState).toMatchSnapshot();
      expect(localStorage.setItem).toHaveBeenCalledWith(
        'accessToken',
        '1234567890'
      );
      expect(localStorage.setItem).toHaveBeenCalledWith('username', 'admin');
    });
  });

  describe('auth/LOGIN_FAILURE', () => {
    it('resets any existing values', () => {
      const state = {
        ...authInitialStates,
        login: {
          ...authInitialStates.login,
          username: 'admin',
          accessToken: '1234567890',
        },
      };
      const newState = authReducers(
        state,
        authOperations.loginFailure({
          message: 'Invalid credentials',
          code: '10014',
        })
      );

      expect(newState).toMatchSnapshot();
    });
  });

  describe('auth/ALREADY_LOGGED_IN', () => {
    beforeEach(() => {
      localStorage.setItem('accessToken', '12345678');
      localStorage.setItem('username', 'admin');
    });

    it('gets localStorage accessToken and username', () => {
      const newState = authReducers(
        authInitialStates,
        authOperations.alreadyLoggedIn()
      );

      expect(newState).toMatchSnapshot();
    });
  });
});
