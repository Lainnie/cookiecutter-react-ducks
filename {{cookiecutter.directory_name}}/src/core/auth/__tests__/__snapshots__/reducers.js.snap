// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth reducers auth/ALREADY_LOGGED_IN gets localStorage accessToken and username 1`] = `
Object {
  "login": Object {
    "accessToken": "12345678",
    "accountId": "",
    "error": false,
    "username": "admin",
  },
}
`;

exports[`auth reducers auth/LOGIN_FAILURE resets any existing values 1`] = `
Object {
  "login": Object {
    "accessToken": "",
    "accountId": "",
    "error": true,
    "username": "",
  },
}
`;

exports[`auth reducers auth/LOGIN_SUCCESS updates username and accessToken 1`] = `
Object {
  "login": Object {
    "accessToken": "1234567890",
    "accountId": "",
    "error": false,
    "username": "admin",
  },
}
`;
